src/map/homunculus.c

```diff
@@ -305,27 +305,46 @@ static int homunculus_skill_tree_get_max(int id, int b_class)
 
 static void homunculus_skillup(struct homun_data *hd, uint16 skill_id)
 {
-	int i = 0 ;
+	int i = 0;
 	nullpo_retv(hd);
 
-	if(hd->homunculus.vaporize != HOM_ST_ACTIVE)
+	if (hd->homunculus.vaporize != HOM_ST_ACTIVE)
 		return;
 
 	i = skill_id - HM_SKILLBASE;
 	Assert_retv(i >= 0 && i < MAX_HOMUNSKILL);
```

Comment:
because ``i`` based on homun skill index from homun skills list, but used in ``skill_tree``, need also check is i valid for ``skill_tree``.

Something like:
``Assert_retv(i >= 0 && i < MAX_HOMUNSKILL && i < MAX_SKILL_TREE);``